### ISSUE TITLE
Build error with @nestjs/swagger 4.0.1

### ISSUE THREAD

**Issue by king-numsgil** [nestjs/swagger#403]
## I'm submitting a...

<pre><code>
[x] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
</code></pre>

## Current behavior

Fails to build with :

```
node_modules/@nestjs/swagger/dist/interfaces/open-api-spec.interface.d.ts:123:5 - error TS2411: Property 'default' of type 'ReferenceObject | ResponseObject | undefined' is not assignable to string index type 'ReferenceObject | ResponseObject'.

123     default?: ResponseObject | ReferenceObject;
        ~~~~~~~
```

Manually editing the error (removing the question mark) fixes the issue, until the package manager runs again or a rebuild is needed (not sure for this one, not verified).

## Expected behavior

Building should not fail with errors outside of my sources

## Minimal reproduction of the problem with instructions

Upgraded `@nestjs/swagger` from 3.1.0 to 4.0.1 and ran `nest start --watch`, which fails.

## Environment

<pre><code>
Nest version: 6.10.5

 
For Tooling issues:
- Node version: 12.13.1  
- Platform: Windows and Linux 
</code></pre>

**Comment by kamilmysliwiec** [comment]
Please, provide a minimal repository which reproduces your issue.

**Comment by king-numsgil** [comment]
[Here it is](https://github.com/king-numsgil/nest-swagger-build-fail)

I found with precision what causes the error : 
```json
	//// THOSE HURT BUILDING SWAGGER
	//"strictNullChecks": true,
	//"strict": true,
	//"alwaysStrict": true
```

Setting any of those to `true` makes the build fail. I prefer to enforce strict types in my projects to reduce bugs a bit...

**Comment by kamilmysliwiec** [comment]
Fixed in 4.0.2 due to #404

**Comment by kamilmysliwiec** [comment]
Also, if you want to use `strict`, I'd recommend using `skipLibCheck` as well
